### PR TITLE
feat(api): implement aggregate functions

### LINKED ISSUE #71: Implement aggregate functionality for users
Create an endpoint for the following functionality:
- Return a `COUNT` of how many movies a user has watched
- Return a `COUNT` of how many lists a user has
- Return a `COUNT` of how many lists have been shared with the user
- Return a `COUNT` of how many comments a user has left

### PR DESCRIPTION
**Pull request by chrispinkney**
## Issue
Closes #71 

## Type of Change

- [ ] **Bugfix**: Change which fixes an issue
- [x] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Description
This PR adds various aggregate functions required by the (yet nonexistent) user page.

## Testing the PR
A new route has been added to postman for user stats testing:
- `get user stats`: `{{BASE_URL}}/users/:id/stats`

## Checklist

- [ ] **Quality**: This PR builds and passes the respective npm run test and works locally
- [ ] **Tests**: This PR includes thorough tests
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

File: api/src/user/dao/user.dao.ts [hunk 1]
```diff
@@ -138,4 +138,73 @@ export class UserDao {
...
+
+        this.prisma.user.findUniqueOrThrow({
+          where: { id: userId },
+          select: {
+            _count: {
+              select: {
+                list: {
+                  where: {
+                    NOT: { creatorId: userId },
+                    user: { some: { id: userId } },
```
Comment (manekenpix) [inline api/src/user/dao/user.dao.ts:168]: We already filter by user in the top `where`, do we need it again?

File: api/src/user/dao/user.dao.ts [hunk 2]
```diff
@@ -138,4 +138,73 @@ export class UserDao {
...
+          },
+        }),
+
+        this.prisma.user.findUniqueOrThrow({
+          where: { id: userId },
+          select: {
+            _count: {
+              select: {
+                movie: {
+                  where: { user: { some: { id: userId } } },
```
Comment (manekenpix) [inline api/src/user/dao/user.dao.ts:182]: We already filter by user in the top `where`, do we need it again?

File: api/src/user/dao/user.dao.ts [hunk 3]
```diff
@@ -138,4 +138,73 @@ export class UserDao {
...
+          },
+        }),
+
+        this.prisma.user.findUniqueOrThrow({
+          where: { id: userId },
+          select: {
+            _count: {
+              select: {
+                comments: {
+                  where: { userId },
```
Comment (manekenpix) [inline api/src/user/dao/user.dao.ts:195]: We already filter by user in the top `where`, do we need it again?

**Review by manekenpix — DISMISSED** [review]
Just a few comments, otherwise good to go.

File: api/src/user/dao/user.dao.ts (hunk 1 above)
Comment (chrispinkney) [reply api/src/user/dao/user.dao.ts:168]: Yes otherwise it doesn't return the proper results.

**Pull request merged**